src/local-state/codegen/visitor.ts

```diff
@@ -0,0 +1,653 @@
+            name
+          );
+          prev[typeName] = applyWrapper(replaced);
+        }
+      }
+
+      return prev;
+    }, {} as ResolverTypes);
+  }
+
+  protected formatRootResolver(
+    schemaTypeName: string,
+    resolverType: string,
+    declarationKind: DeclarationKind
+  ): string {
+    const avoidOptionals = this.config.avoidOptionals.resolvers;
+
+    return `${schemaTypeName}${
+      avoidOptionals ? "" : "?"
+    }: ${resolverType}${this.getPunctuation(declarationKind)}`;
+  }
+
+  private clearOptional(str: string): string {
+    if (str.startsWith("Maybe")) {
+      return str.replace(/Maybe<(.*?)>$/, "$1");
+    }
+
+    return str;
+  }
+
+  ListType(node: ListTypeNode): string {
+    return `Maybe<${super.ListType(node)}>`;
+  }
+
+  DirectiveDefinition(
+    _node: DirectiveDefinitionNode,
+    _key: string | number,
+    _parent: any
+  ): string {
+    return "";
+  }
+
+  EnumTypeDefinition(node: EnumTypeDefinitionNode): string {
+    const rawTypeName = node.name as any;
+
+    // If we have enumValues set, and it's point to an external enum - we need to allow internal values resolvers
+    // In case we have enumValues set but as explicit values, no need to to do mapping since it's already
+    // have type validation (the original enum has been modified by base types plugin).
+    // If we have mapper for that type - we can skip
+    if (
+      !this.config.mappers[rawTypeName] &&
+      !this.config.enumValues[rawTypeName]
+    ) {
+      return "";
+    }
+
+    const name = this.convertName(node, {
+      suffix: this.config.resolverTypeSuffix,
+    });
+    this._collectedResolvers[rawTypeName] = {
+      typename: name,
+      baseGeneratedTypename: name,
+    };
+    const hasExplicitValues = this.config.enumValues[rawTypeName]?.mappedValues;
+
+    return new DeclarationBlock(this._declarationBlockConfig)
+      .export()
+      .asKind("type")
+      .withName(name)
+      .withContent(
+        hasExplicitValues ?
+          this.buildEnumResolversExplicitMappedValues(
+            node,
+            this.config.enumValues[rawTypeName].mappedValues!
+          )
+        : this.buildEnumResolverContentBlock(
+            node,
+            this.getTypeToUse(rawTypeName)
+          )
+      ).string;
+  }
+
+  ObjectTypeDefinition(node: ObjectTypeDefinitionNode): string {
+    const declarationKind = "type";
+    const name = this.convertName(node, {
+      suffix: this.config.resolverTypeSuffix,
+    });
+    const typeName = node.name as any as string;
+
+    const rootType = ((): false | "query" | "mutation" | "subscription" => {
+      if (this.schema.getQueryType()?.name === typeName) {
+        return "query";
+      }
+      if (this.schema.getMutationType()?.name === typeName) {
+        return "mutation";
+      }
+      if (this.schema.getSubscriptionType()?.name === typeName) {
+        return "subscription";
+      }
+      return false;
+    })();
```

Comment:
This would be a bit easier for me to read, but tbh., I don't know if it's even an improvement.
```suggestion
    const rootType =
      {
        [this.schema.getQueryType()?.name || "__"]: "query",
        [this.schema.getMutationType()?.name || "__"]: "mutation",
        [this.schema.getSubscriptionType()?.name || "__"]: "subscription",
      }[typeName] || false;
```

Reply:
Ah ya that was some copy/paste from the `BaseResolversVisitor`: https://github.com/dotansimha/graphql-code-generator/blob/a4acc9e80fafe232fa161ebe8b1d784765127a8d/packages/plugins/other/visitor-plugin-common/src/base-resolvers-visitor.ts#L1645-L1656. I had to override that function so I copied the implementation and tweaked what I needed, but didn't pay much attention to how the rest of it was written.

Reply:
Fine with me :)